Mathlib/Geometry/Manifold/IntegralCurve.lean

```diff
@@ -501,4 +500,33 @@ theorem isIntegralCurve_Ioo_eq_of_contMDiff_boundaryless [BoundarylessManifold I
     (hγ : IsIntegralCurve γ v) (hγ' : IsIntegralCurve γ' v) (h : γ t₀ = γ' t₀) : γ = γ' :=
   isIntegralCurve_eq_of_contMDiff (fun _ ↦ BoundarylessManifold.isInteriorPoint I) hv hγ hγ' h
 
+/-- A (global) integral curve is injective iff it is not periodic. -/
+lemma IsIntegralCurve.periodic_iff_not_injective [BoundarylessManifold I M]
+    (hγ : IsIntegralCurve γ v)
+    (hv : ContMDiff I I.tangent 1 (fun x => (⟨x, v x⟩ : TangentBundle I M))) :
+    (∃ T > 0, Periodic γ T) ↔ ¬Injective γ := by
```

Comment:
Please prove a more specific statement: if `γ t = γ t'`, then `γ` is periodic with period `t' - t`.

Reply:
Sorry, forgot to address your other review comment. `M` needs to be boundaryless because the uniqueness theorem is only stated for interior points at the moment. We likely need submanifolds to address the case involving points on the boundary.

Reply:
It should be possible to avoid this assumption, but let's postpone it until later.
bors merge

Reply:
I agree. It's on my TODO. Thanks for merging.